File: src/main/java/io/pivotal/security/repository/InMemorySecretStore.java
package io.pivotal.security.repository;

import io.pivotal.security.entity.NamedSecret;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

// TODO this is no longer necessary
@Component
public class InMemorySecretStore {

  private final InMemorySecretRepository secretRepository;

  @Autowired
  public InMemorySecretStore(InMemorySecretRepository secretRepository) {
    this.secretRepository = secretRepository;
  }

  public void set(NamedSecret secret) {
    secretRepository.save(secret);
  }

  public NamedSecret getSecret(String key) {
    return secretRepository.findOneByName(key);
  }

  public boolean delete(String key) {
    NamedSecret entity = secretRepository.findOneByName(key);
    if (entity != null) {
      secretRepository.delete(entity);
      return true;
    }
    return false;
  }
}

File: src/main/kotlin/io/pivotal/security/repository/InMemorySecretStore.kt
package io.pivotal.security.repository

import io.pivotal.security.entity.NamedSecret
import io.pivotal.security.model.Secret
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
open class InMemorySecretStore
@Autowired
constructor(private val secretRepository: InMemorySecretRepository) : SecretStore {

  @Transactional
  override fun set(key: String, secret: Secret) {
    val namedSecret: NamedSecret = secretRepository.findOneByName(key) ?: NamedSecret(name = key)
    namedSecret.type = secret.type!!
    namedSecret.value = secret.value
    secretRepository.save<NamedSecret>(namedSecret)
  }

  override fun get(key: String): Secret? {
    val namedSecret = secretRepository.findOneByName(key)
    if (namedSecret != null) {
      return Secret(namedSecret.value, namedSecret.type)
    }
    return null
  }

  @Transactional
  override fun delete(key: String): Secret? {
    val namedSecret = secretRepository.findOneByName(key)
    if (namedSecret != null) {
      secretRepository.delete(namedSecret)
      return Secret(namedSecret.value, namedSecret.type)
    }
    return null
  }

}
